server/datastore/mysql/activities.go

```diff
@@ -231,14 +231,31 @@ func (ds *Datastore) MarkActivitiesAsStreamed(ctx context.Context, activityIDs [
 // software to install, etc.) and provides a unified view of those upcoming
 // tasks.
 func (ds *Datastore) ListHostUpcomingActivities(ctx context.Context, hostID uint, opt fleet.ListOptions) ([]*fleet.Activity, *fleet.PaginationMetadata, error) {
+	// NOTE: Be sure to update both the count (here) and list statements (below)
+	// if the query condition is modified.
```

Comment:
ah damn, good catch. Is there any way we could write a test to catch this? not necessarily in this PR

we probably didn't catch this because the expectation is hardcoded? so something like `require.Equal(t, len(activities), count)`?

Reply:
I was thinking about it, I think the hard-coded expectation is ok in the sense that it didn't catch it because the hard-coded value was wrong. It would be hard to not hard-code it and catch issues with missing conditions because we also need to make sure in the test that we create execution requests that match the new condition.